cluster/shard_mapper.go

```diff
@@ -84,42 +108,49 @@ func NewRemoteMaper(nodeID, shardID uint64, stmt string, chunkSize int) *RemoteM
 }
 
 // Open connects to the remote node and starts receiving data.
-func (r *RemoteMapper) Open() error {
-	node, err := r.MetaStore.Node(r.nodeID)
+func (r *RemoteMapper) Open() (err error) {
+	defer func() {
+		if err != nil {
+			r.conn.MarkUnusable()
```

Comment:
Very good -- I see what you mean. This is not correct. 

Let me fix that, so that we mark connections as unusable only under the right circumstances.